lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -567,15 +569,10 @@ Ufe::UndoableCommand::Ptr UsdContextOps::doOpCmd(const ItemPath& itemPath)
         return std::make_shared<SetVariantSelectionUndoableCommand>(prim(), itemPath);
     } // Variant sets
     else if (itemPath[0] == kUSDToggleVisibilityItem) {
-        auto attributes = Ufe::Attributes::attributes(sceneItem());
-        assert(attributes);
-        auto visibility = std::dynamic_pointer_cast<Ufe::AttributeEnumString>(
-            attributes->attribute(UsdGeomTokens->visibility));
-        assert(visibility);
-        auto current = visibility->get();
-        return visibility->setCmd(
-            current == UsdGeomTokens->invisible ? UsdGeomTokens->inherited
-                                                : UsdGeomTokens->invisible);
+        auto object3d = UsdObject3d::create(fItem);
+        auto current = object3d->visibility();
```

Comment:
Shouldn't we check that the scene item supports the Object3d interface?  At this point it should, because we put up the menu entry for it, so perhaps we can just assert.

Reply:
> A couple of things to fix, but really what I feel is missing from this PR are the automated tests. You can write an automated test that first tests the UsdUndoVisibleCommand, and calls undo and redo on that object, for the lowest-level test.

Great point. I am working on it as we speak.

Addressed in 70046317e72410a8a661614b42f579146509cba3

Reply:
@ppt-adsk  @seando-adsk  

Added a test for backing the new UsdUndoVisibleCommand :

865d4793f94be4b42021a90c3f416164c2868cdd